torch/testing/_internal/common_methods_invocations.py

```diff
@@ -146,6 +146,14 @@ def formatter(arg):
 
         return self._repr_helper(formatter)
 
+    def log(self):
+        """
+        Print the name of the SampleInput to stdout. This is supposed to be called
+        by tests, to make it possible to know which SampleInput was in use in
+        case of a test failure
+        """
+        print(f'[SampleInput] {self.name}')
```

Comment:
I think printing only on failure is the killer feature we want to go for with this PR. 

See https://github.com/pytorch/pytorch/issues/60748, I think it'd be generally useful.

Reply:
I tried to design a solution in PR #61910